llama_hub/tools/metaphor/base.py

```diff
@@ -0,0 +1,104 @@
+"""Metaphor tool spec."""
+
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from typing import Optional, List
+from llama_index.readers.schema.base import Document
+import requests
+import urllib.parse
+import datetime
+
+
+class MetaphorToolSpec(BaseToolSpec):
+    """Metaphor tool spec."""
+
+    spec_functions = [
+        "search",
+        "retrieve_documents",
+        "current_date",
+        "find_similar",
+    ]
+
+    def __init__(self, api_key: str) -> None:
+        """Initialize with parameters."""
+        from metaphor_python import Metaphor
+
+        self.client = Metaphor(api_key=api_key)
+
+    def search(
+        self,
+        query: str,
+        num_results: Optional[int] = 10,
+        include_domains: Optional[List[str]] = None,
+        exclude_domains: Optional[List[str]] = None,
+        start_published_date: Optional[str] = None,
+        end_published_date: Optional[str] = None,
+    ) -> str:
+        """
+        Metaphor allows you to use a natural language query to search the internet
+
+        Args:
+            query (str): A natural language query phrased as an answer for what the link provides, ie: "This is the latest news about space:"
+            num_results (Optional[int]): Number of results to return. Defaults to 10.
+            include_domains (Optional[List(str)]): A list of top level domains like ["wsj.com"] to limit the search to specfic sites.
+            exclude_domains (Optional[List(str)]): Top level domains to exclude.
+            start_published_date (Optional[str]): A date string like "2020-06-15". Get the date from `current_date`
+            end_published_date (Optional[str]): End date string
+        """
+        response = self.client.search(
+            query,
+            num_results=num_results,
+            include_domains=include_domains,
+            exclude_domains=exclude_domains,
+            start_published_date=start_published_date,
+            end_published_date=end_published_date,
+            use_autoprompt=True
+        )
+        return [
+            {"title": result.title, "url": result.url, "id": result.id}
+            for result in response.results
+        ]
+
+    def retrieve_documents(self, ids: List[str]) -> List[Document]:
+        """
+        Retrieve a list of document summaries returned by `metaphor_search`, using the ID field
+
+        args:
+            ids (List(str)): the ids of the documents to retrieve
+        """
+
+        response = self.client.get_contents(ids)
+        return [Document(text=con.extract) for con in response.contents]
+
+    def find_similar(
+        self,
+        url: str,
+        num_results: Optional[int] = 3,
```

Comment:
i see you changed the other value to 10. do you want this to be 10 too? personally i'm not sure of the merits of 3 vs 10. 3 may be better if there's no filtering being done